keps/83-workload-preemption/README.md

```diff
@@ -0,0 +1,449 @@
+  // disrupting other Workloads.
+  //
+  // 0 <= cost <= 100
+  Cost int32
+}
+```
+
+#### LocalQueue changes
+
+The new API fields in LocalQueue describe the defaults that apply to Workloads
+in this LocalQueue.
+
+```golang
+type LocalQueueSpec struct {
+  ...
+  // preemptionDefaults are the defaults applied to Workloads assigned to this
+  // LocalQueue. The defauts are applied to Workloads at the time of the
+  // Workload admission.
+  PreemptionDefaults *WorkloadPreemption
+}
+```
+
+#### ClusterQueue changes
+
+The new API fields in ClusterQueue describe how to influence the selection
+of Workloads to preempt.
+
+```golang
+const ClusterQueuePendingWorkloadsUnderMinQuota = "PendingWorkloadsUnderMinQuota"
+
+type ClusterQueueSpec struct {
+  ...
+  // preemption describes policies to preempt Workloads from ClusterQueues
+  // borrowing resources when this ClusterQueue has pending Workloads and the
+  // min quota is unmet.
+  preemption ClusterQueuePreemption
+}
+
+type ClusterQueueStatus struct {
+  ...
+  conditions []metav1.Condition
+}
+
+type ClusterQueuePreemption struct {
+  // waitBeforePreemptionSeconds is the time in seconds that the ClusterQueue
+  // is willing to wait for Workloads to finish before issuing preemptions,
+  // since the first time that the ClusterQueue has pending workloads that would
+  // fit under the min quota. This time is present in the timestamp for the
+  // PendingWorkloadsUnderMinQuota condition.
+  // If null, it means to wait indefinitely.
+  WaitBeforePreemptionSeconds *int64
+
+  // <UNRESOLVED>
+
+  // allowPreemptionByLowerPriority determines whether pending Workloads can
+  // preempt Workloads with higher priority in this ClusterQueue.
+  // Defaults to false.
+  AllowPreemptionByLowerPriority bool
+
+  // workloadSorting determines how Workloads from the cohort that are
+  // candidates for preemption are sorted.
+  // Sorting happens at the time when a Workload in this ClusterQueue is
+  // evaluated for admission. All the Workloads in the cohort are sorted based
+  // on the criteria defined in the preempting ClusterQueue.
+  // workloadOrder is a list of comparison criteria between two Workloads that
+  // are evaluated in order.
+  // Possible criteria are:
+  // - ByLowestPriority: Prefer to preempt the Workload with lower priority.
+  // - ByLowestCost: Prefer to preempt the workload with lower preemption cost.
+  // - ByLowestRuntime: Prefer to preempt the Workload that started more
+  //   recently.
+  // - ByLongestRuntime: Prefer to preempt the Workload that started earlier.
+  //
+  // If empty, the behavior is equivalent to
+  // [ByLowestPriority, ByLowestCost, ByLowestRuntime].
+  WorkloadSorting []WorkloadComparison
+
+  type WorkloadSortingCriteria string
+
+  const (
+    ComparisonByLowestPriority = "ByLowestPriority"
+    ComparisonByLowestCost = "ByLowestCost"
+    ComparisonByLowestRuntime = "ByLowestRuntime"
+  )
```

Comment:
I don't think this is necessary, at least not initially. The default of priority -> cost -> runtime looks very reasonable and for me it matches the semantics of priority and cost.